tests/ui/parser/missing-fat-arrow.rs

```diff
@@ -0,0 +1,41 @@
+fn main() {
+    let x = 1;
+    let y = 2;
+    let value = 3;
+
+    match value {
+        Some(x) if x == y {
+            self.next_token()?; //~ ERROR expected identifier, found keyword `self`
+            Ok(true)
+        },
+        _ => {
+            Ok(false)
+        }
+    }
+    let _: i32 = (); //~ ERROR mismatched types
```

Comment:
How is this relevant?

Reply:
To ensure that the parser recovers and eventually gets to typeck.